server/src/main/java/io/crate/types/GeoShapeType.java

```diff
@@ -124,14 +124,13 @@ public Map<String, Object> sanitizeValue(Object value) {
 
     @Override
     public int compare(Map<String, Object> val1, Map<String, Object> val2) {
-        // TODO: compare without converting to shape
-        Shape shape1 = GeoJSONUtils.map2Shape(val1);
-        Shape shape2 = GeoJSONUtils.map2Shape(val2);
-        return switch (shape1.relate(shape2)) {
-            case WITHIN -> -1;
-            case CONTAINS -> 1;
-            default -> Double.compare(shape1.getArea(JtsSpatialContext.GEO), shape2.getArea(JtsSpatialContext.GEO));
-        };
+        if (Maps.equals(val1, val2)) {
+            return 0;
+        } else {
+            // Order is undefined for GEO SHAPES, returning arbitrary value.
+            // TODO: Add a method to the DataType indicating comparison support and use it instead of SemanticSortValidator.SUPPORTED_TYPES.
+            return 1;
+        }
```

Comment:
```suggestion
        return MapComparator.compareMaps(val1, val2);
```

Reply:
Thanks, I was not aware that we have it already

Reply:
> I don't see any Array specific logic here

`Objects.deepEquals` has that logic.

On MapsComparator: Tests indeed are passing but if I revert change done in `test_cast_wkt_point_string_array_to_geo_shape_array` test fails with 

`java.lang.IllegalArgumentException: Cannot detect the type of the value: [D@32ab68`
(it cannot work with a primitive).

I think it's still fine as we don't expect primitives and it's only artificially coming from the assertion?
See also https://github.com/crate/crate/pull/16761#discussion_r1795478418

Reply:
Our type system doesn't support primitive types as values, such we must ensure to always use non-primitive values when we want to detect or sanitize them. This can be fixed by parsing GeoShape coordinates as `Double[]` instead of `double[]`. I've pushed a commit demonstrating this suggestion.

> Fix DataType.guessType to deal with Double[] (it's not hitting case Object[] array).

This won't help/work as the parsed GeoShape coordinates we want to compare with will still return `double[]`

 > Avoid type guessing in the MapsComparator.

Reply:
Thanks for the suggestion. While fixing a test with size estimations (test_bytes_estimate has some entries with a bigger expected value due to Double Object overhead).

However, some entries got lower which was surprusing. I investigated a bit, looks like we are not hitting primitive array based logic introduced in 
https://github.com/crate/crate/commit/90f19dbe54241d0dfb21ed44b859831c5d2bb04b, I will fix it as well

Reply:
Pushed an adjustment of geo shape accounting https://github.com/crate/crate/pull/16761/commits/ec87d239423a8830a7272e49a593406f06db2897